packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -0,0 +1,1340 @@
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<a href="#" {{action "edit" bubbles=false}}>click me</a>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('a').click();
+    });
+
+    this.assert.ok(editHandlerWasCalled, 'the child event handler was called');
+    this.assert.notOk(originalHandlerWasCalled, 'the parent handler was not called');
+  }
+
+  ['@test it should allow "send" as the action name (#594)']() {
+    let sendHandlerWasCalled = false;
+
+    let ExampleComponent = Component.extend({
+      actions: {
+        send() { sendHandlerWasCalled = true; }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<a href="#" {{action "send"}}>click me</a>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('a').click();
+    });
+
+    this.assert.ok(sendHandlerWasCalled, 'the event handler was called');
+  }
+
+  ['@test it should send the view, event, and current context to the action']() {
+    let passedTarget;
+    let passedContext;
+
+    let aTarget = Component.extend({
+      actions: {
+        edit(context) {
+          passedTarget = this;
+          passedContext = context;
+        }
+      }
+    }).create();
+
+    let aContext;
+
+    let ExampleComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        aContext = this;
+      },
+      aTarget
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<a id="edit" href="#" {{action "edit" this target=aTarget}}>click me</a>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('#edit').click();
+    });
+
+    this.assert.strictEqual(passedTarget, aTarget, 'the action is called with the target as this');
+    this.assert.strictEqual(passedContext, aContext, 'the parameter is passed along');
+  }
+
+  ['@test it should only trigger actions for the event they were registered on']() {
+    let editHandlerWasCalled = false;
+
+    let ExampleComponent = Component.extend({
+      actions: {
+        edit() { editHandlerWasCalled = true; }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<a href="#" {{action "edit"}}>click me</a>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('a').click();
+    });
+
+    this.assert.ok(editHandlerWasCalled, 'the event handler was called on click');
+
+    editHandlerWasCalled = false;
+
+    this.runTask(() => {
+      this.$('a').trigger('mouseover');
+    });
+
+    this.assert.notOk(editHandlerWasCalled, 'the event handler was not called on mouseover');
+  }
+
+  ['@skip should unwrap controllers passed as a context']() {
```

Comment:
I think that we don't support this any more (because proxying controllers are removed).